layer/layer.go

```diff
@@ -188,6 +188,8 @@ type Store interface {
 	GetRWLayer(id string) (RWLayer, error)
 	GetMountID(id string) (string, error)
 	ReleaseRWLayer(RWLayer) ([]Metadata, error)
+	LockRWLayer()
+	UnlockRWLayer()
```

Comment:
I agree, this is not a suitable addition to the interface.

If `TarStream` needs to block other operations it should hold a lock that blocks other operations until it is completed the stream.